### ISSUE TITLE
GetPackageOriginalFile errors bsdtar: Error opening archive: Unrecognized archive format

### ISSUE THREAD

**Issue by OJFord** [CyberShadow/aconfmgr#100]
For example:
```
rm /var/cache/pacman/pkg/glibc*
GetPackageOriginalFile glibc /etc/locale.gen
```
gives me:
> bsdtar: Error opening archive: Unrecognized archive format

but manually running the same from:
https://github.com/CyberShadow/aconfmgr/blob/e2e1164d3fce60491fa65da99c7c52a8927a9d2b/src/common.bash#L1495
```
bsdtar -x --to-stdout --file /var/cache/pacman/pkg/glibc-2.33-5-x86_64.pkg.tar.zst etc/locale.gen
```

lists the file contents as expected, with no error, exiting 0.

**Comment by CyberShadow** [comment]
It works for me. Can you put together a reproducer (e.g. using Docker)?

**Comment by OJFord** [comment]
With `set -x` on, I can see it's caused by trying to read the signature file instead of the actual tar:
```
++++ bsdtar -x --to-stdout --file /var/cache/pacman/pkg/glibc-2.33-5-x86_64.pkg.tar.zst.sig etc/locale.gen
bsdtar: Error opening archive: Unrecognized archive format
```

A naive patch changing:
https://github.com/CyberShadow/aconfmgr/blob/11df29ed264b1236a134345acc16c8e1ab59b483/src/common.bash#L1309

to `"%q-%q-%q.pkg.tar.zst"` fixes it; broken by 1176b833bd8dcbdbc4e7b7a5022c697f8fd0e3ca (or perhaps the `.sig`s started being cached since then, I don't know).

However I note from 790726e9a48752a4097d4866238c7c63575788a1 that this is too naive. Need either to whitelist endings like `.xz`, `.zst`, and whatever else; or specifically blacklist `.sig`.

**Comment by CyberShadow** [comment]
Thanks for that analysis. I don't see any `.sig` files at all, maybe because of pacman 5.

> or specifically blacklist `.sig`.

That seems the most appropriate to me too.
